kivy/tools/packaging/pyinstaller_hooks/rt-hook-kivy.py

```diff
@@ -29,15 +29,17 @@
 environ['KIVY_MODULES_DIR'] = join(root, 'modules')
 environ['KIVY_EMBED'] = '1'
 
+
 # Monkey-patch pygame to get around an issue with Pygame window icon and
 # PyInstaller 2.1. See kivy issue #1638
-# Uncomment the following to package pygame
-#import pygame.pkgdata
-#_original_getResource = pygame.pkgdata.getResource
-#
-#
-#def getResource(identifier, *args, **kwargs):
-#    if identifier == 'pygame_icon.tiff':
-#        raise IOError()
-#    return _original_getResource(identifier, *args, **kwargs)
-#pygame.pkgdata.getResource = getResource
+def getResource(identifier, *args, **kwargs):
```

Comment:
How does this load to a crash at runtime? That's why it's under a
try/except block.
On Apr 26, 2015 6:21 AM, "Akshay Arora" <notifications@github.com> wrote:

> In kivy/tools/packaging/pyinstaller_hooks/rt-hook-kivy.py
> <https://github.com/kivy/kivy/pull/3303#discussion_r29108450>:
>
> >  # Monkey-patch pygame to get around an issue with Pygame window icon and
> >  # PyInstaller 2.1. See kivy issue #1638
> > -# Uncomment the following to package pygame
> > -#import pygame.pkgdata
> > -#_original_getResource = pygame.pkgdata.getResource
> > -#
> > -#
> > -#def getResource(identifier, *args, **kwargs):
> > -#    if identifier == 'pygame_icon.tiff':
> > -#        raise IOError()
> > -#    return _original_getResource(identifier, *args, **kwargs)
> > -#pygame.pkgdata.getResource = getResource
> > +def getResource(identifier, *args, **kwargs):
>
> @matham <https://github.com/matham> we don't provide pygame anymore,
> there is not reason to look to patch pygame at runtime and this does lead
> to a crash for a packaged app using just SDL2.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kivy/kivy/pull/3303/files#r29108450>.
>